spring-boot/proxy/src/main/java/eu/coatrack/proxy/filters/pre/ApiKeyAuthFilter.java

```diff
@@ -54,14 +55,31 @@
     public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
         log.debug(String.format("Api key auth filter received request of type %s", servletRequest.getClass()));
 
+        Authentication authToken = null;
         if (servletRequest instanceof HttpServletRequest) {
             final HttpServletRequest req = (HttpServletRequest) servletRequest;
             final String keyParam = req.getParameter(ApiKey.API_KEY_REQUEST_PARAMETER_NAME);
             log.debug(String.format("request parameter '%s' has value '%s'", ApiKey.API_KEY_REQUEST_PARAMETER_NAME, keyParam));
 
-            Authentication authToken = new ApiKeyAuthToken(keyParam, Collections.emptyList());
+            authToken = new ApiKeyAuthToken(keyParam, Collections.emptyList());
             SecurityContextHolder.getContext().setAuthentication(authToken);
         }
         filterChain.doFilter(servletRequest, servletResponse);
+
+        if (SecurityContextHolder.getContext().getAuthentication() == null)
+            causeInternalServerErrorWhenAuthenticationCouldNotBeDecided(authToken);
+    }
+
+    private void causeInternalServerErrorWhenAuthenticationCouldNotBeDecided(Authentication authToken) {
+        Authentication resultAuthentication;
+        try {
+            resultAuthentication = apiKeyAuthenticator.authenticate(authToken);
+        } catch (Exception e) {
+            //Exception is already logged by the AuthenticationManager.
+            return;
+        }
+        if (resultAuthentication == null)
+            throw new AuthenticationValidityCouldNotBeDecidedException("The Gateway is not able to decide the " +
```

Comment:
Why is the message divided in 2 strings, wouldn't be better if this was only on one String?

Reply:
There is a maximum on how long a line of code should be. When you read code, you don't have to scroll far to the right. The String was separated in order not to exceed this limit.

Reply:
I was now thinking if it is not better to put it on the final static String and call the string here